internal/locate/region_request.go

```diff
@@ -1046,6 +1071,7 @@ func (s *RegionRequestSender) SendReqCtx(
 			return nil, nil, err
 		}
 		if regionErr != nil {
+			lastRegionErr = regionErrorToLabel(regionErr)
 			retry, err = s.onRegionError(bo, rpcCtx, req, regionErr)
```

Comment:
We may need to collect information about retry reason of region error details for each `replicaSelectorRound`.